src/client/java/teammates/client/scripts/scalabilitytests/InstructorFeedbackResultsPageScalabilityTest.java

```diff
@@ -18,28 +24,145 @@
 
     private static final String DATA_FOLDER_PATH = "src/client/java/teammates/client/scripts/scalabilitytests/data/";
     private static final Logger log = Logger.getLogger();
+    private static DataBundle testDataMax;
 
     @Override
     protected void prepareTestData() {
         // the actual test data is refreshed before each test method
     }
 
     private void refreshTestData(int numStudents, int numQuestions) {
-        refreshTestData(
-                "InstructorFeedbackResultsPageScaleTest-" + numStudents + "Students" + numQuestions + "Questions.json");
-    }
 
-    private void refreshTestData(String filename) {
-        testData = loadDataBundle(DATA_FOLDER_PATH + filename);
+        // modify set of students for each test according to its requirements
+        if (testData.students.size() == 0) { // if current collection of students is empty
+            addStudents(numStudents);
+        } else if (testData.students.size() < numStudents) { // current number of students is below required
+            addStudents(numStudents - testData.students.size());
+        } else if (testData.students.size() > numStudents) { // current number of students is above required
+            decreaseNumOfStudents(numStudents);
+        }
+
+        // modify set of questions for each test according to its requirements
+        if (testData.feedbackQuestions.size() == 0) { // if current collection of questions is empty
+            addQuestions(numQuestions);
+        } else if (testData.feedbackQuestions.size() < numQuestions) { // current number of questions is below required
+            addQuestions(numQuestions - testData.feedbackQuestions.size());
+        } else if (testData.feedbackQuestions.size() > numQuestions) { // current number of questions is above required
+            decreaseNumOfQuestions(numQuestions);
+        }
+
+        // modify set of responses for each test according to its sets of students and questions
+        updateFeedbackResponses();
+
         removeAndRestoreDataBundle(testData);
```

Comment:
This PR is neat; unfortunately it doesn't do anything to reduce the number of Datastore operations at all because this line still does the remove and restore operation on the entire `testData` bundle.

To correct, the approach has to be changed. Do a persist (without remove) data bundle for adding new data (this should be what the test measures). At the very end of the test, do a remove data bundle for all data. You'll need to add some methods in `BaseTestCaseWithBackDoorApiAccess` to support this.

EDIT: I just realised the total number of operations would still be the same, even with that approach. @samsontmr I'm not sure how you intended to reduce the number?